cfg/lock.go

```diff
@@ -45,9 +47,46 @@ func (lf *Lockfile) WriteFile(lockpath string) error {
 	return ioutil.WriteFile(lockpath, o, 0666)
 }
 
+// Clone returns a clone of Lockfile
+func (lf *Lockfile) Clone() *Lockfile {
+	n := &Lockfile{}
+	n.Hash = lf.Hash
+	n.Updated = lf.Updated
+	n.Imports = lf.Imports.Clone()
+	n.DevImports = lf.DevImports.Clone()
+
+	return n
+}
+
+// Fingerprint returns a hash of the contents minus the date. This allows for
+// two lockfiles to be compared irrespective of their updated times.
+func (lf *Lockfile) Fingerprint() (string, error) {
```

Comment:
If you use `sha256.Sum256` instead of using the `hash.Hasher` interface, you can return a `[32]byte` or `*[32]byte` instead of a slice.  Arrays of equal size can be compared with `==` so there's no need to use `bytes.Equal` for them.